### PR TITLE
feat(device_info_plus): add major, minor and patch versions to macos

### PR DESCRIPTION
**Pull request by goenning**
## Description

This PR adds major/minor/patch versions to mac info class, similar to what's available on windows.

With this fields we don't need to parse the version from of the osRelease field (which apple doesn't recommend).

https://developer.apple.com/documentation/foundation/operatingsystemversion

## Checklist

- [X] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [X] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [X] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [X] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by goenning** [comment]
Android build is failing, but doesn't seem related?

**Comment by vbuberen** [comment]
Thanks for your contribution.

I would also ask you to add these new fields to the `example` app, so they will be visible as well near that `osRelease` string.

> Android build is failing, but doesn't seem related?

Yes, correct. Android tests are flaky here. But seems like re-run helped.

**Comment by goenning** [comment]
Done! :)
Screenshot 2023-04-04 at 22 05 00

**Comment by nohli** [comment]
Should we also return a String with all, like version: 13.1.0?

**Comment by goenning** [comment]
I thin that's something the user can easily concatenate on their side. It's the same with windows info, there are 3 separate fields.

File: packages/device_info_plus/device_info_plus/macos/Classes/DeviceInfoPlusMacosPlugin.swift [hunk 1]
```diff
@@ -18,12 +18,17 @@ public class DeviceInfoPlusMacosPlugin: NSObject, FlutterPlugin {
     }
 
     private func handleDeviceInfo(result: @escaping FlutterResult)-> Void{
+        let osVersion = ProcessInfo.processInfo.operatingSystemVersion;
+
```
Comment (nohli) [inline packages/device_info_plus/device_info_plus/macos/Classes/DeviceInfoPlusMacosPlugin.swift:22]: Is the whitespace needed?

```suggestion
        let osVersion = ProcessInfo.processInfo.operatingSystemVersion;
```

Comment (goenning) [reply packages/device_info_plus/device_info_plus/macos/Classes/DeviceInfoPlusMacosPlugin.swift]: Removed, and also moved the osVersion closer to where it's used

**Review by nohli — APPROVED** [review]
LGTM!

**Comment by goenning** [comment]
Thakns @nohli I couldn't find any info on the release process. How often does the package get released to pub.dev?

**Comment by nohli** [comment]
Right now, there is a bug in our publishing tool, but vbueren will try to publish later today.

**Pull request merged**